pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1185,6 +1226,63 @@ func parseUnderlayNetworkConfig(appInstance *types.AppInstanceConfig,
 	}
 }
 
+func checkUnderlayNetworkForDuplicatePortMap(ulCfgList0 []types.UnderlayNetworkConfig,
+	ulCfgList1 []types.UnderlayNetworkConfig) bool {
+
+	for _, ulCfg0 := range ulCfgList0 {
+		for _, ulCfg1 := range ulCfgList1 {
+			// not on the same network instance
+			if ulCfg0.Network != ulCfg1.Network {
```

Comment:
I might be wrong, but I think the same uplink can be shared by multiple network instances. If that is the case we should compare across network instances also.

Reply:
makes sense, will add changes to reflect this behavior also.